ultimate_tools.py

```diff
@@ -0,0 +1,137 @@
+from typing import List, Tuple, Dict, Union
+
+from src import fastq_tools
+from src import dna_rna_tools
+from src import protein_tools
+
+
+def filter_fastq_seqs(
+        seqs: Dict[str, Tuple[str]],
+        gc_bounds: Union[float, Tuple[float]] = (0, 100),
+        length_bounds: Union[int, Tuple[int]] = (0, 2 ** 32),
+        quality_threshold: int = 0
+) -> Dict[str, Tuple[str]]:
+    """
+    Filters provided sequences with specified parameters.
+
+    Args:
+    - seqs (Dict[str, Tuple[str]]): a dictionary with FASTQ-file contents
+    - gc_bounds (Union[float, Tuple[float]]): GC-content boundaries (from 0 to 100)
+        within which filtered sequences must be included
+    - length_bounds (Union[int, Tuple[int]]): length boundaries
+        within which filtered sequences must be included
+    - quality_threshold (int): reading quality value according to table phred+33,
+        below which filtering will not be performed
+
+    Return:
+    - Dict[str, Tuple[str]]: a dictionary with filtered sequences
+    """
+
+    if isinstance(gc_bounds, (int, float)):  # input check
```

Comment:
Кстати, забавный факт, с python3.10 можно еще и так писать:
```suggestion
    if isinstance(gc_bounds, int | float):  # input check
```